### PR TITLE
Scrit test

### PR DESCRIPTION
**Pull request by anand-1mg**
Deputy dev test changes.

---

 **Size XL:** This PR changes include 634 lines and should take approximately 3-6 hours to review

---

 **DeputyDev generated PR summary:** 

The provided Pull Request (PR) involves several changes, primarily related to the configuration and setup of a Python project using IntelliJ IDEA (or a similar IDE). Here's a breakdown of what this PR does:

1. **IDE Configuration Files**: 
   - Adds several configuration files under the `.idea` directory, which is typically used by IntelliJ IDEA to store project-specific settings.
   - These files include `.gitignore` for ignoring specific folders like `shelf/` and `workspace.xml`, `dbnavigator.xml` for database navigation settings, `hector.iml` for module settings, `inspectionProfiles` for code inspection settings, and `modules.xml` for module management.
   - The `vcs.xml` file sets up version control system mappings, indicating the use of Git.

2. **Python Script (`bot2.py`)**:
   - Introduces a new Python script named `bot2.py` located in the `hector` directory.
   - This script defines a `CoverageBot` class that is responsible for parsing coverage reports (`diff-coverage.json` and `coverage.json`), formatting comments based on these reports, and posting comments to a Bitbucket pull request.
   - The script uses several modules like `rich` for console output and `os` for environment variable access.
   - It includes functionality for connecting to Bitbucket's API to post comments about code coverage, using an `APIClient`.
   - The script is designed to be run as a standalone program, with a `main` function that initializes the `CoverageBot` and posts a comment.

Overall, this PR sets up the necessary infrastructure for a Python project within an IDE environment, focusing on code coverage reporting and interaction with Bitbucket for automated comments on pull requests.

---

 **Size XL:** This PR changes include 634 lines and should take approximately 3-6 hours to review

---

 **DeputyDev generated PR summary:** 

The pull request (PR) introduces several changes primarily related to the integration of IDE-specific configurations and a new Python script (`bot2.py`) for handling code coverage reporting. Here's a breakdown of the changes:

1. **IDE Configuration Files**:
   - Adds several `.idea` directory files, which are configuration files for JetBrains IDEs like PyCharm. These include:
     - `.gitignore`: Specifies files and directories to be ignored by version control within the `.idea` directory.
     - `dbnavigator.xml`: Contains database navigation and settings configurations.
     - `hector.iml`: Module configuration for the project, indicating it's a Python module.
     - `inspectionProfiles/Project_Default.xml` and `profiles_settings.xml`: Define code inspection settings and profiles.
     - `modules.xml`: Lists modules associated with the project.
     - `vcs.xml`: Configures version control settings, specifically mapping the directory to use Git.

2. **Python Script (`bot2.py`)**:
   - Introduces a new script `bot2.py` that implements a `CoverageBot` class. This class is responsible for:
     - Parsing coverage reports from JSON files (`diff-coverage.json` and `coverage.json`).
     - Generating and posting comments to a Bitbucket pull request, detailing the code coverage statistics.
     - Utilizing the `rich` library for formatting console output and comments.
     - Handling API requests to post comments using the `APIClient` class.
     - The script also includes a main execution block for testing the bot functionality.

The primary purpose of this PR is to set up the project with IDE-specific configurations and add functionality to automate coverage report commenting on Bitbucket pull requests. There are no direct code improvements or technical discussions in the PR diff itself, but it provides a foundational setup for a development environment and a tool for enhancing code review processes through automated coverage reporting.

### CODE REVIEW

File: hector/bot2.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+        if not diff_cov or not cov:
+            diff_cov, cov = self._parse_reports()
+        self.diff_cov = diff_cov
+        self.cov = cov
+
+    def post(
+        self,
+        dry: Optional[bool] =  False
+    ):
+        with console.status("Preparing Comment"):
```
Comment (deepaks-1mg) [inline hector/bot2.py:29]: #dd - Can you suggest a modular code for this method?

Comment (deepaks-1mg) [inline hector/bot2.py:29]: #dd - Can you suggest a modular code for this method?

**Review by deepaks-1mg — CHANGES_REQUESTED** [review]

File: hector/bot2.py (hunk 1 above)
Comment (deepaks-1mg) [inline hector/bot2.py:29]: #dd - Can you suggest. modular code for this?

File: hector/bot2.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
+import json
+import os
+
+from datetime import datetime, timedelta, timezone
+from typing import Optional, Union
+
+from rich.panel import Panel
+from rich.pretty import Pretty
+
+from .client import APIClient
```
Comment (deepaks-1mg) [inline hector/bot2.py:10]: #dd - Can you suggest a modular code here?

**Review by deepaks-1mg — APPROVED** [review]

File: .idea/.gitignore [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# Default ignored files
+/shelf/
+/workspace.xml
```
Comment (deepaks-1mg) [inline .idea/.gitignore:3]: #dd - Can you provide feedback on this?

File: hector/bot2.py [hunk 4]
```diff
@@ -0,0 +1,168 @@
+import json
```
[the remark this answers is not in the document]
Comment (ankit27755) [reply hector/bot2.py:1]: #dd explain this

File: hector/bot2.py [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+import os
+
+from datetime import datetime, timedelta, timezone
```
Comment (ankit27755) [inline hector/bot2.py:4]: #dd what is this line doing? Explain

File: .idea/.gitignore [hunk 6]
```diff
@@ -0,0 +1,3 @@
+# Default ignored files
+/shelf/
```
Comment (deepaks-1mg) [inline .idea/.gitignore:2]: #dd - Can you provide feedback on this gitignore?

File: hector/bot2.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (ankit27755) [reply hector/bot2.py:1]: #dd what does this means?

File: hector/bot2.py [hunk 7]
```diff
@@ -0,0 +1,168 @@
+import json
+import os
```
Comment (ankit27755) [inline hector/bot2.py:2]: #dd what does this signifies?

File: hector/bot2.py [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+    with open("diff-coverage.json", encoding="utf-8") as f:
+        diff_cov = DiffCovReport(**json.load(f))
+
+    with open("coverage.json", encoding="utf-8") as f:
+        cov = CovReport(**json.load(f))
+
+    bot = CoverageBot(cov=cov, diff_cov=diff_cov)
+
+    bot_comment = bot.get_comment()
+    bot_logger.info(bot_comment)
```
Comment (anand-1mg) [inline hector/bot2.py:167]: #dd what is this line doing bot_logger.info(bot_comment)?

File: .idea/.gitignore (hunk 3 above)
Comment (deepaks-1mg) [inline .idea/.gitignore:3]: #dd - Can you suggest feedback on these changes?

Comment (deepaks-1mg) [inline .idea/.gitignore:3]: #dd - Can you suggest feedback on these changes?

File: .idea/inspectionProfiles/profiles_settings.xml [hunk 9]
```diff
@@ -0,0 +1,6 @@
+<component name="InspectionProjectProfileManager">
+  <settings>
+    <option name="USE_PROJECT_PROFILE" value="false" />
+    <version value="1.0" />
```
Comment (anand-1mg) [inline .idea/inspectionProfiles/profiles_settings.xml:4]: #dd what is this use?

File: .idea/hector.iml [hunk 10]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module type="PYTHON_MODULE" version="4">
+  <component name="NewModuleRootManager">
+    <content url="file://$MODULE_DIR$" />
+    <orderEntry type="inheritedJdk" />
```
Comment (anand-1mg) [inline .idea/hector.iml:5]: #dd what is this change?

File: .idea/.gitignore (hunk 3 above)
Comment (anand-1mg) [reply .idea/.gitignore:3]: #dd - Can you suggest feedback on these changes?

File: .idea/modules.xml [hunk 11]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="ProjectModuleManager">
+    <modules>
+      <module fileurl="file://$PROJECT_DIR$/.idea/hector.iml" filepath="$PROJECT_DIR$/.idea/hector.iml" />
+    </modules>
```
Comment (anand-1mg) [inline .idea/modules.xml:6]: #dd what is this change?

File: .idea/inspectionProfiles/profiles_settings.xml [hunk 12]
```diff
@@ -0,0 +1,6 @@
+<component name="InspectionProjectProfileManager">
+  <settings>
```
Comment (anand-1mg) [inline .idea/inspectionProfiles/profiles_settings.xml:2]: #dd what is this pr doing?

File: .idea/inspectionProfiles/profiles_settings.xml (hunk 9 above)
Comment (anand-1mg) [reply .idea/inspectionProfiles/profiles_settings.xml:4]: #dd what is this change?

File: hector/bot2.py [hunk 13]
```diff
@@ -0,0 +1,168 @@
...
+import os
+
+from datetime import datetime, timedelta, timezone
+from typing import Optional, Union
+
+from rich.panel import Panel
+from rich.pretty import Pretty
+
+from .client import APIClient
+from .console import console
```
Comment (anand-1mg) [inline hector/bot2.py:11]: #dd what is this line doing?

File: .idea/.gitignore (hunk 3 above)
[the remark this answers is not in the document]
Comment (anand-1mg) [reply .idea/.gitignore:3]: #dd what is this line doing?

File: hector/bot2.py [hunk 14]
```diff
@@ -0,0 +1,168 @@
...
+            file_stats = self._format_file_stats()
+            comment += file_stats
+
+        # add footer
+        commit_link = self._format_commit_link()
+        timestamp = self.timestamp()
+        comment += f"###### 📝 Reported on commit {commit_link} | {timestamp})"
+        return comment
+
+    def _title_emoji(self):
```
Comment (anand-1mg) [inline hector/bot2.py:85]: #dd what is this change?

File: hector/bot2.py [hunk 15]
```diff
@@ -0,0 +1,168 @@
...
+        if self.returncode!=0:
+            return "❗️"
+        return "✅"
+
+    def _format_file_stats(self):
+        src_stats = self.diff_cov.src_stats
+        file_stats = ["```sh"]
+        for file, stat in src_stats.items():
+            if stat["percent_covered"] < 100:
```
Comment (anand-1mg) [inline hector/bot2.py:94]: #dd what is this change?

**Pull request closed without merging**